docs/source/federation/metrics.md

```diff
@@ -0,0 +1,37 @@
+
+> NOTE: By default, metrics will be reported to the `'current'` variant. To change the variant for reporting, set the `ENGINE_GRAPH_VARIANT` environment variable.
+
+## How tracing data is exposed from a federated service
+
+The Apollo gateway looks to the `extensions` field of all service responses for the presence of an `ftv1` field. This field contains a representation of the tracing information for the sub-query that was executed against the service, sent as the base64 encoding of the [protobuf representation](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting-protobuf/src/reports.proto) of the trace. By default, a federated Apollo Server service will emit this tracing information in its extension if sent the header pair `'apollo-federation-include-trace': 'ftv1'`. By setting the gateway's API key via the environment variable, it will send this header pair by default as well.
+
+## How the gateway constructs and reports traces
+
+The Apollo gateway constructs traces in the shape of the [query plan](https://www.apollographql.com/docs/apollo-server/federation/implementing/#inspecting-query-plans), embedding an individual `Trace` at each fetch node to indicate the sub-query traces and in which order they were fetched from the underlying services. The field-level statistics that the metrics ingress aggregates from these traces are collected over the fields over which the operation was executed **in the implementing services**. In other words, field stats are collected based on the operations the query planner makes rather than the operations that the clients make. On the other hand, query-level statistics are aggregated over the operations exected **by the client**, which means that even if query-planning changes, statistics will still be corresponded to the same client-delivered operation.
+
+## How errors work
+
+The Apollo Platform provides functionality to modify error details for the client, via the [formatError](https://www.apollographql.com/docs/apollo-server/features/errors/#for-the-client-response) option. Additionally, there is functionality to support modifying error details for the metrics ingress, via the [rewriteError](https://www.apollographql.com/docs/apollo-server/features/errors/#for-apollo-engine-reporting) option.
+
+For the former, modifying errors for the client, you might want to use this option to hide implementation details, like Database errors, from your users. For the latter, modifying errors for reporting, you might want to obfuscate or redact personal information, like user IDs or emails.
```

Comment:
I might just say _When modifying errors for the client_, rather than _also_ using the _Former_ specifier.